### PR TITLE
[WEB-4041] 14.6.8 - pricing card title tooltips, support for multiple tooltip orientations

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

https://ably.atlassian.net/jira/software/projects/WEB/boards/172?selectedIssue=WEB-4041

Motivated by this request here: https://ably-real-time.slack.com/archives/C019XP5SYB0/p1728566555191379?thread_ts=1728019882.765889&cid=C019XP5SYB0

## Summary of changes

- adds the ability to have tooltips in pricing card titles
- adds relevant ones into the default dataset for consumption cards
- adds support for interactive tooltips with multiple orientations

## How do you manually test this?

https://ably.github.io/ably-ui/?path=/story/features-pricing-cards--consumption

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Summary by CodeRabbit

- **New Features**
	- Introduced tooltip functionality for titles in the Pricing Cards, enhancing user interaction with additional contextual information.
	- Added detailed tooltips for "Channels" and "Connections" in pricing data, linking to further documentation.

- **Improvements**
	- Enhanced positioning logic for tooltips, allowing for better responsiveness and interaction based on cursor movements.

- **Version Update**
	- Updated project version from 14.6.7 to 14.6.8.

### CODE REVIEW

File: src/core/Tooltip.tsx [hunk 1]
```diff
@@ -98,14 +102,24 @@ const Tooltip = ({
...
+
+      if (orientation === "top") {
+        return clientX >= x && clientX <= x + width && clientY < y;
+      } else if (orientation === "left") {
+        return clientY >= y && clientY <= y + width && clientX < x;
+      } else if (orientation === "right") {
+        return clientY >= y && clientY <= y + width && clientX > x + width;
+      } else if (orientation === "bottom") {
+        return clientX >= x && clientX <= x + width && clientY > y + width;
+      }
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Tooltip.tsx:122]: The icon is square, but it's good to be explicit - will change

**Review by kennethkalmer — APPROVED** [review]
Looks good in Safari 👍

File: src/core/Tooltip.tsx [hunk 2]
```diff
@@ -98,14 +102,24 @@ const Tooltip = ({
...
     event: MouseEvent,
     ref: RefObject<HTMLButtonElement>,
   ) => {
     if (ref.current) {
       const { clientX, clientY } = event;
-      const { x, y, width } = ref.current.getBoundingClientRect();
-      return clientX >= x && clientX <= x + width && clientY < y;
+      const { x, y, width, height } = ref.current.getBoundingClientRect();
+      const { orientation } = position;
+
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Tooltip.tsx:113]: Sure, good suggestion

**Comment by jamiehenson** [comment]
I added [a bit extra](https://github.com/ably/ably-ui/pull/507/files#diff-f4d05aae27a488d9bde520f2863c5e45a5d15b2599a18e192248797f7d706cf4R131-R145) to try and make interactive hovers a bit more stable - by determining whether the cursor is within the original triggering element upon leaving the tooltip after a short time has passed. Not 100% but it helps

**Review by aralovelace — APPROVED** [review]
⭐⭐⭐⭐⭐

**Pull request merged**